tools/chapel-py/method-tables.h

```diff
@@ -0,0 +1,461 @@
+/*
+ * Copyright 2021-2023 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//
+// This file defines nearly all the methods exposed to python from the frontend.
+//
+// To use these definitions, define the macro and then `#include` this file.
+// See `chapel.cpp` for examples. If the macros below are not defined, they are
+// defined to be empty. Only the ones that are used at the time of include need
+// to be defined.
+//
+
+//
+// Defines the beginning of an ast node
+//
+#ifndef METHOD_TABLE_BEGIN
+#define METHOD_TABLE_BEGIN(TAG)
+#endif
+
+//
+// Defines the end of an ast node
+//
+#ifndef METHOD_TABLE_END
+#define METHOD_TABLE_END(TAG)
+#endif
+
+//
+// Defines a simple getter that takes no arguments
+//
+// BODY is an inline definition of the getter. Available variables:
+// - contextObject: the current frontend context
+// - node: the ast node
+// TYPESTR is the return type. Possible values:
+// - "O" = python object
+// - "s" = string
```

Comment:
Since you're writing docs ( 😄 ) might be good to mention that "s" can be any of (C string, C++ string, unique string).